webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/WCIFHandler.java

```diff
@@ -0,0 +1,195 @@
+package net.gnehzr.tnoodle.server.webscrambles;
+
+import static net.gnehzr.tnoodle.utils.GsonUtils.GSON;
+import static net.gnehzr.tnoodle.utils.GwtSafeUtils.azzert;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+public class WCIFHandler {
+    
+    // Currently, we mark not cubing related activities as other-lunch or other-speech, for example.
+    // If we ever accept any other such ignorable key, it should be added here.
+    private static final String[] wcifIgnorableKeys = new String[]{"other"};
+
+    private ArrayList<ScrambleRequest> allScrambleRequests;
+    private JsonObject schedule;
+    private JsonParser parser;
+
+    public WCIFHandler(LinkedHashMap<String, String> query) {
```

Comment:
I think this class should instead take a `ScrambleRequest[]` or `ArrayList<ScrambleRequest>`, and the string/query parsing should happen up in a `*Handler` class.

Reply:
Fixed